frontend/src/metabase/admin/people/components/SearchInput.jsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled, { css } from "styled-components";
+import { color } from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+import { t } from "ttag";
+
+export default function SearchInput({
+  value,
+  placeholder,
+  onChange,
+  className,
+  hasClearButton,
+}) {
+  const handleClearClick = () => {
+    onChange("");
+  };
+
+  const showClearButton = hasClearButton && value.length > 0;
+
+  return (
+    <SearchFieldRoot className={className}>
+      <IconWrapper>
+        <Icon name="search" size={16} />
+      </IconWrapper>
+      <Input
+        hasClearButton
+        type="text"
+        placeholder={placeholder}
+        value={value}
+        onChange={e => onChange(e.target.value)}
+      />
+
+      {showClearButton && (
+        <ClearButton onClick={handleClearClick}>
+          <Icon name="close" size={12} />
+        </ClearButton>
+      )}
+    </SearchFieldRoot>
+  );
+}
+
+SearchInput.propTypes = {
+  onChange: PropTypes.func.isRequired,
+  placeholder: PropTypes.string,
+  value: PropTypes.string,
+  autoFocus: PropTypes.bool,
+  className: PropTypes.string,
+  hasClearButton: PropTypes.bool,
+};
+
+SearchInput.defaultProps = {
+  placeholder: t`Find...`,
+  value: "",
+  autoFocus: false,
+  hasClearButton: false,
+};
+
+const Input = styled.input`
```

Comment:
@kdoh I was not sure if we have a defined Input in our design system. There are multiple inputs at the moment including:
- `frontend/src/metabase/components/Input.jsx` — this one has different styles and it uses `Button` css classes inside
- `frontend/src/metabase/components/InputBlurChange.jsx`
- `frontend/src/metabase/admin/datamodel/components/FormInput.jsx` — form input which is currently expects `field` form object
- The one is used inside `AccordionList`
- Some custom `<input type="text" ... />`

I'd love to reuse an existing one but I did not found one. I think we need to unify our UI, improve our UI kit and discourage creating custom components in place. I haven't done it here because premature unification or limiting abstraction without knowing all requirements, use cases for the component is worse than creating a new isolated input in place. What do you think?